kuka_rsi_hw_interface/krl/README.md

```diff
@@ -1,4 +1,115 @@
+6. Add a new IP address on another subnet (e.g. 192.168.1.20) for the **RSI interface**.
+   * Select the entry **New** under **RSI Ethernet** in the tree structure and press **Edit**.
+   * Enter the IP address and confirm with **OK**.
+   * Close **RSI-Network** and maximize the SmartHMI.
+7. Reboot the controller with a cold restart (**Shutdown > Check *Force cold start* and *Reload files* > Reboot control PC**).
+8. After reboot, minimize the SmartHMI (**Start-up > Service > Minimize HMI**).
+9. Run **cmd.exe** and ping the PC you want to communicate with on the same subnet (e.g. 192.168.250.xx).
+
+If your **PC** has an IP address on the same subnet as the **Windows interface** on the controller, the controller should receive answers from the PC:
+* If this is the case, add another IP address to the current PC connection (e.g. 192.168.1.xx) on the same subnet as the **RSI** interface.
+
+## 2. KRL Files
+
+The files included in this folder specifies the data transferred via RSI. Some of the files needs to be modified to work for your specific configuration.
+
+##### ros_rsi_ethernet.xml
+1. Edit the `IP_NUMBER` tag so that it corresponds to the IP address (192.168.1.xx) previously added for your PC.
+2. Keep the `PORT` tag as it is (49152) or change it if you want to use another port.
+
+Note that the `rsi/address` and `rsi/port` parameters of the `kuka_rsi_hw_interface` must correspond to the `IP_NUMBER`and `PORT` set in these KRL files.
+
+##### ros_rsi.rsi.xml
+This file may be edited with application specific joint limits in degrees.
+* Edit the parameters within the RSIObject `AXISCORR` to specify joint limits such as **LowerLimA1**, **UpperLimA1** etc. Note that these limits are in reference to the start position of the robot.
+* Edit the parameters within `AXISCORRMON` to specify the overall correction limitation. If this limit is exceeded in either of the joint directions, RSI is stopped. The values of **MaxA1**, **MaxA2** etc. may be large to allow free movement within the specified joint limits in `AXISCORR`.
+
+Notice the RSIObject of type `ETHERNET`. Within this object is a parameter called **Timeout**. This parameter is set to **100** by default. The RSI interface operates at `250 Hz` (4ms cycle). The **kuka_rsi_hardware_interface** does not have a period configured and is completely driven by the controller's output. Every controller RSI output has a IPOC timestamp which increments for every cycle. The **kuka_rsi_hardware_interface** will answer as quickly as possible. The answer includes the last IPOC received. If the connected **PC with ROS** did not manage to answer within the RSI cycle of **4ms**, the IPOC timestamp of RSI has incremented. The IPOC included in the answer is not matched and the controller will increment a counter. When this counter hits the **Timeout** parameter (**100**), the RSI connection will shut down. If this parameter is lowered, the demand for real-time computation will increase.
+

```

Comment:
From what I can tell, 250 Hz is actually the default.

From the 3.2 manual:

> The signal processing is calculated at the sensor cycle rate. Depending on the mode, the
sensor cycle rate is 12 ms (IPO mode) or 4 ms (IPO_FAST mode).

and

> The signal processing is carried out by default in IPO_FAST mode. In this
case, the reference coordinate system for the sensor correction must be con-
figured in the RSI object POSCORR. If the signal processing is activated in
IPO mode, the reference coordinate system must be defined with RSI_ON().

![screenshot from 2016-08-30 08 35 00](https://cloud.githubusercontent.com/assets/9865385/18090970/ae518114-6e8c-11e6-9d16-9ee728d6d832.png)